algorithms/C/searching/linearsearch.c

```diff
@@ -0,0 +1,29 @@
+/* Program to search data in given array using linear search */
+#include<stdio.h>
+int main()
+{
+    int len;
+    printf("Enter the lenght of array");
+    scanf("%d",&len); //scanning the legth of array
+}
```

Comment:
you are trying to write your code outside the main without any user-defined function which shows an error 
so create a function name linear_search and call it from main